web/src/components/Friendship/FriendRequests.jsx

```diff
@@ -0,0 +1,127 @@
+import { useAuth0 } from "@auth0/auth0-react";
+import { useState, useEffect } from "react";
+import * as React from "react";
+import Api from "../../helpers/api";
+import Button from "../Common/Button";
+import Popup from "../Common/Popup";
+
+const Friendrequests = () => {
+  const [requests, setRequests] = useState([]);
+  const [friends, setFriends] = useState([]);
+  const { getAccessTokenSilently } = useAuth0();
+
+  const api = new Api();
+
+  useEffect(() => {
+    try {
+      console.log("getting friend requests");
+      getAllFriendRequests();
+    } catch (err) {
+      console.log("error" + err);
+    }
+  }, []);
+
+  useEffect(() => {
+    try {
+      console.log("getting friend information");
+      getFriendInformation();
+    } catch (err) {
+      console.log("error" + err);
+    }
+  }, [requests]);
+
+  async function getAllFriendRequests() {
+    await api.getFriendRequests(await getAccessTokenSilently()).then((res) => {
+      setRequests(res.friendRequests);
+    });
+  }
+
+  async function getFriendInformation() {
+    const friendsTemp = [];
+    for (let i = 0; i < requests.length; i++) {
+      console.log("friend # " + i);
+      console.log(requests[i].sending_profile_id);
+      try {
+        api
+          .getFriendInformation(
+            await getAccessTokenSilently(),
+            requests[i].sending_profile_id
+          )
+          .then((res) => {
+            console.log(res);
+            friendsTemp[i] = res.data.profileFriend[0];
+            setFriends(friendsTemp);
+            console.log(friends);
+          });
+      } catch (err) {
+        console.log("error" + err);
+      }
+    }
+  }
+
+  async function declineFriendRequest(id) {
+    console.log("Declining Request");
+    console.log(id);
+
+    //delete event with specified id
+    api.deleteFriendRequest(await getAccessTokenSilently(), id).then(() => {
+      console.log("Deleting Friend Request");
+      getAllFriendRequests();
+    });
+  }
+
+  async function acceptFriendRequest(friendID) {
+    console.log("Accepting Request");
+
+    //create friendship with specified id
+    api
+      .createFriendship(await getAccessTokenSilently(), friendID)
+      .then((res) => {
+        console.log(res);
+        console.log("Creating Friend Request");
+      });
+
+    console.log("id: " + friendID);
+
+    //delete event with specified id
+    api
+      .deleteFriendRequest(await getAccessTokenSilently(), friendID)
+      .then(() => {
+        console.log("Deleting Friend Request");
+        getAllFriendRequests();
+      });
```

Comment:
For page to reload automatically after accepted invitation, add:  window.location.reload();